### PR TITLE
X7S support

### PR DESCRIPTION
**Pull request by 3djc**
- remove X7S option
- runtime detection of bluetooth chip (detection result can be checked in 3rd statistic screen)

There is now a single binary for X7 or X7S, BT handling is made at runtime.

Many thanks to @Kilrah for his support :)

### CODE REVIEW

File: radio/src/gui/128x64/view_statistics.cpp [hunk 1]
```diff
@@ -117,6 +117,7 @@ void menuStatisticsView(event_t event)
   #define MENU_DEBUG_COL2_OFS          (17*FW)
```
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

File: radio/src/gui/128x64/view_statistics.cpp [hunk 2]
```diff
@@ -117,6 +117,7 @@ void menuStatisticsView(event_t event)
...
   #define MENU_DEBUG_Y_CURRENT         (1*FH)
```
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

File: radio/src/gui/128x64/view_statistics.cpp [hunk 3]
```diff
@@ -117,6 +117,7 @@ void menuStatisticsView(event_t event)
...
   #define MENU_DEBUG_ROW1              (1*FH+1)
```
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

File: radio/src/gui/128x64/view_statistics.cpp [hunk 4]
```diff
@@ -117,6 +117,7 @@ void menuStatisticsView(event_t event)
...
+  #define MENU_DEBUG_ROW2              (2*FH+1)
```
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

File: radio/src/gui/128x64/view_statistics.cpp [hunk 5]
```diff
@@ -117,6 +117,7 @@ void menuStatisticsView(event_t event)
...
   #define MENU_DEBUG_Y_MAH             (2*FH)
```
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

File: radio/src/gui/128x64/view_statistics.cpp [hunk 6]
```diff
@@ -117,6 +117,7 @@ void menuStatisticsView(event_t event)
...
   #define MENU_DEBUG_Y_CPU_TEMP        (3*FH)
```
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

File: radio/src/gui/128x64/view_statistics.cpp [hunk 7]
```diff
@@ -117,6 +117,7 @@ void menuStatisticsView(event_t event)
...
   #define MENU_DEBUG_Y_COPROC          (4*FH)
```
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

**Comment by opentx-sonar** [comment]
SonarQube analysis reported 23 issues
* ![CRITICAL][CRITICAL] 17 critical
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L32](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L32): Add a "default" case to this switch statement. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ASwitchWithoutDefault)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L116](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L116): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L124](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L124): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L125](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L125): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L126](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L126): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L127](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L127): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L128](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L128): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L130](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L130): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L137](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L137): Add a "default" case to this switch statement. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ASwitchWithoutDefault)
1. ![CRITICAL][CRITICAL] [opentx.cpp#L2096](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/opentx.cpp#L2096): Either add a parameter list or the "&" operator to this use of "menuRadioDiagAnalogs". [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS936)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by opentx-sonar** [comment]
SonarQube analysis reported 21 issues
* ![CRITICAL][CRITICAL] 17 critical
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [radio_hardware.cpp#L49](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/radio_hardware.cpp#L49): Add a "default" case to this switch statement. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ASwitchWithoutDefault)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L32](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L32): Add a "default" case to this switch statement. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ASwitchWithoutDefault)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L116](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L116): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L124](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L124): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L125](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L125): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L126](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L126): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L127](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L127): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L128](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L128): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L130](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L130): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L137](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L137): Add a "default" case to this switch statement. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ASwitchWithoutDefault)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by opentx-sonar** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [opentx.cpp#L409](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/opentx.cpp#L409): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1066)
1. ![MAJOR][MAJOR] [opentx.cpp#L2888](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/opentx.cpp#L2888): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1066)
1. ![MINOR][MINOR] [opentx.cpp#L1389](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/opentx.cpp#L1389): Extract this nested code block into a separate method. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1199)
1. ![MINOR][MINOR] [opentx.cpp#L1672](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/opentx.cpp#L1672): Remove this empty statement. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1116)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: radio/src/gui/128x64/view_statistics.cpp (hunk 1 above)
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

File: radio/src/gui/128x64/view_statistics.cpp (hunk 2 above)
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

File: radio/src/gui/128x64/view_statistics.cpp (hunk 3 above)
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

File: radio/src/gui/128x64/view_statistics.cpp (hunk 4 above)
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

File: radio/src/gui/128x64/view_statistics.cpp (hunk 5 above)
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

File: radio/src/gui/128x64/view_statistics.cpp (hunk 6 above)
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

File: radio/src/gui/128x64/view_statistics.cpp (hunk 7 above)
Comment (opentx-sonar) [inline radio/src/gui/128x64/view_statistics.cpp]: CRITICAL Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

**Comment by opentx-sonar** [comment]
SonarQube analysis reported 24 issues
* ![CRITICAL][CRITICAL] 20 critical
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [radio_hardware.cpp#L138](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/radio_hardware.cpp#L138): Refactor this function to reduce its Cognitive Complexity from 73 to the 25 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS3776)
1. ![CRITICAL][CRITICAL] [radio_hardware.cpp#L154](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/radio_hardware.cpp#L154): Add a "default" case to this switch statement. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ASwitchWithoutDefault)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L32](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L32): Add a "default" case to this switch statement. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3ASwitchWithoutDefault)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L116](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L116): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L124](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L124): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L125](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L125): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L126](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L126): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L127](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L127): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L128](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L128): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)
1. ![CRITICAL][CRITICAL] [view_statistics.cpp#L130](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/gui/128x64/view_statistics.cpp#L130): Remove the whitespaces before the # character. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3APPDirectiveIndentation)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by opentx-sonar** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [opentx.cpp#L409](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/opentx.cpp#L409): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1066)
1. ![MAJOR][MAJOR] [opentx.cpp#L2888](https://github.com/opentx/opentx/blob/a9b02378b75fbd4375acedc08eedf6861fafe957/radio/src/opentx.cpp#L2888): Merge this if statement with the enclosing one. [rule](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1066)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by bsongis — APPROVED** [review]

**Pull request merged**